cpp/include/cudf/utilities/span.hpp

```diff
@@ -132,6 +132,16 @@ struct host_span : public span_base<T, Extent, host_span<T, Extent>> {
   constexpr host_span(C const& in) : base(in.data(), in.size())
   {
   }
+
+  template <typename OtherT,
+            std::size_t OtherExtent,
+            typename std::enable_if<(Extent == OtherExtent || Extent == dynamic_extent) &&
+                                      std::is_convertible<OtherT(*), T(*)>::value,
```

Comment:
I thought pointer to array syntax require the square brackets? I can't find a description of this syntax anywhere.

Reply:
For some reason it did not work with square brackets. The case I want to enable is simple <int> to <int const>, and somehow that did not pass with brackets. I can share a godbolt sample.

Reply:
Huh, now it seems to work: https://godbolt.org/z/K3osEr
I'll try again in our code.

Reply:
Works fine now, not sure what went wrong initially.